### PR TITLE
EZEE-3479: As a developer I want to have reusable componenet for charts

### PR DESCRIPTION
**Pull request by lucasOsti**
| Question      | Answer
| ------------- | ---
| Tickets       | https://issues.ibexa.co/browse/EZEE-3479
| Related to | https://github.com/ezsystems/ezplatform-personalization/pull/176
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | no
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Added reusable component for chart

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                ];
+            }
+
+            return XAxes;
+        }
+
+        getScalesYAxesOption() {
+            let YAxes = [];
+
+            if (this.type === 'line') {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:50]: `this.type` this is never set in this base class and this class definitely should not know what are the values of this properties

**Review by dew326 — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+        setCheckboxBackground(checkbox) {
+            const { checkedColor } = checkbox.dataset;
+            const { checked } = checkbox;
+
+            if (checked) {
+                checkbox.style.backgroundColor = checkedColor;
+                checkbox.style.borderColor = checkedColor;
+            } else {
+                checkbox.style.backgroundColor = '#fff';
```
Comment (GrabowskiM) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:25]: I would put colors to constants at the beginning of file

**Review by GrabowskiM — CHANGES_REQUESTED** [review]

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+        layoutOptions() {}
+
+        scaleOptions() {}
+
+        legendOptions(chart) {}
+
+        render() {
+            this.chart = new Chart(this.canvas.getContext('2d'), {
+                type: this.getType(),
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:75]: `getType` is not defined in this class

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                data: {
+                    labels: this.labels,
+                    datasets: this.datasets,
+                },
+                options: {
+                    responsive: true,
+                    maintainAspectRatio: false,
+                    layout: this.layoutOptions(),
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:83]:
```suggestion
                    layout: this.getLayoutOptions(),
```
?

File: src/bundle/Resources/public/js/scripts/core/base.chart.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                        onComplete: (animation) => {
+                            const chart = animation.chart;
+                            const chartMethod = chart.config.data.datasets.length ? 'remove' : 'add';
+                            const chartNode = chart.canvas.closest('.ez-chart');
+
+                            chartNode.dispatchEvent(new CustomEvent('ez-chart-animation-complete'));
+                            chartNode.classList[chartMethod]('ez-chart--no-data');
+                        },
+                    },
+                    scales: this.scaleOptions(),
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/base.chart.js:123]:
```suggestion
                    scales: this.getScaleOptions(),
```
?

File: src/bundle/Resources/public/js/scripts/core/line.chart.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+                                return value;
+                            },
+                        },
+                    },
+                ],
+            };
+        }
+
+        legendOptions(chart) {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/core/line.chart.js:81]:
```suggestion
        getLegendOptions(chart) {
```
?

**Review by dew326 — CHANGES_REQUESTED** [review]

**Review by dew326 — APPROVED** [review]

**Review by tomaszszopinski — APPROVED** [review]
QA approved on Ibexa dxp 3.3.1-dev with patch.

**Pull request merged**